### ISSUE TITLE
Debian package should use absolute Qdrant directory

### ISSUE THREAD

**Issue by timvisee** [qdrant/qdrant#3370]
**Is your feature request related to a problem? Please describe.**
In <https://github.com/qdrant/qdrant/pull/3344>, we've introduced a Debian package for Qdrant. It installs `qdrant` on your system.

Starting `qdrant` will use the current working directory for configuration, storage and snapshots. This can be problematic.

**Describe the solution you'd like**
Instead, Qdrant should use a absolute path that, just like similar tools. That way we ensure that Qdrant files are always in the same place, no matter from what working directory we start it from.

We should store everything in `/var/lib/qdrant` and `/etc/qdrant` by default, so that we get:
- `/var/lib/qdrant/storage`
- `/var/lib/qdrant/snapshots`
- `/var/lib/qdrant/static` (for web UI [files](https://github.com/qdrant/qdrant/blob/master/tools/sync-web-ui.sh))
- `/etc/qdrant/config.yaml`

Other suggestions are welcome as well.

**Additional context**
Mimicking defaults other similar tools have is likely a good idea here.

**Comment by shankariraja** [comment]
Addressing #3370: Updated configuration file; testing in progress. Open to feedback and suggestions!

**Comment by ai-naymul** [comment]
> Addressing #3370: Updated configuration file; testing in progress. Open to feedback and suggestions!

Don't forget to update the documentation also :)

**Comment by Snehil-Shah** [comment]
@timvisee I tried installing and running qdrant using the deb package, but the dashboard is not working. Apparently it can't find the ./static directory. Am I doing something wrong? The REST API is working fine
![image](https://github.com/qdrant/qdrant/assets/130062020/1974464d-cef9-4f6e-ad7e-0112a007ce17)

**Comment by Pegasis404** [comment]
Does setting path using env variables will be good solution to this?

**Comment by timvisee** [comment]
> Does setting path using env variables will be good solution to this?

It would be best to ship the [configs](https://github.com/qdrant/qdrant/tree/master/config) and store them in `/var/lib/qdrant/config/`. In `config.yaml` we could change the default paths to be absolute, as listed in the first comment.

For this we do need to extend Qdrant a bit to be able to load this configuration file. There is a `--config-file <PATH>` flag, but I think we should use a different approach.

We can change [this](https://github.com/qdrant/qdrant/blob/bb5e3b9e10b5798164f8e62e525af1ec9ea065bb/src/main.rs#L100C9-L100C9) to `#[arg(long, value_name = "PATH"), env = "QDRANT_CONFIG_PATH"]`. Then we can set `QDRANT_CONFIG_PATH=/var/lib/qdrant/config/config.yaml`.

So to summarize:
- store [configs](https://github.com/qdrant/qdrant/tree/master/config) in `/var/lib/qdrant/config/`
- main config should have absolute paths
- update argument parsing to source config path from `QDRANT_CONFIG_PATH`
- set `QDRANT_CONFIG_PATH=/var/lib/qdrant/config/config.yaml`

**Comment by bikeshedder** [comment]
For Debian packages it's also quite common to ship with a file in `/etc/default` file. That could be used to pass extra start-up arguments to the binary. As qdrant is building its own debian package that won't be necessary unless there are command line arguments that can not or should not be part of the configuration file.

**Comment by timvisee** [comment]
@bikeshedder Thank you for your insights! Putting it in `/etc/qdrant` makes sense. I'll update the original issue description to reflect this.

**Comment by timvisee** [comment]
Fixed by <https://github.com/qdrant/qdrant/pull/6958>

**Comment by timvisee** [comment]
[Qdrant 1.15.2](https://github.com/qdrant/qdrant/releases/tag/v1.15.2) has been released, which includes the fix for this problem.
